src/platform/datapath_raw_xdp.c

```diff
@@ -1077,88 +1077,74 @@ CxPlatDpRawInitialize(
         Xdp,
         Xdp->WorkerCount);
 
-    PIP_ADAPTER_ADDRESSES Adapters = NULL;
-    ULONG Error;
-    ULONG AdaptersBufferSize = 15000; // 15 KB buffer for GAA to start with.
-    ULONG Iterations = 0;
-    ULONG flags = // skip info that we don't need.
-        GAA_FLAG_INCLUDE_PREFIX |
-        GAA_FLAG_SKIP_UNICAST |
-        GAA_FLAG_SKIP_ANYCAST |
-        GAA_FLAG_SKIP_MULTICAST |
-        GAA_FLAG_SKIP_DNS_SERVER |
-        GAA_FLAG_SKIP_DNS_INFO;
-
-    do {
-        Adapters = (IP_ADAPTER_ADDRESSES*)CxPlatAlloc(AdaptersBufferSize, ADAPTER_TAG);
-        if (Adapters == NULL) {
+    PMIB_IF_TABLE2 pIfTable;
+    if (GetIfTable2(&pIfTable) != NO_ERROR) {
+        Status = QUIC_STATUS_INTERNAL_ERROR;
+        goto Error;
+    }
+
+    CXPLAT_FRE_ASSERT(pIfTable->NumEntries <= 128);
+    BOOLEAN Initialized[128] = {0};
```

Comment:
I'm a bit unsure, but do we _really_ need this array? I don't like the hard coded limit, but it seems more like an optimization that's not really necessary.

Reply:
I want to use std::bitset or heap allocated array. It is possible to remove this by checking Xdp->Interfaces linked list at beginning of outer loop whether there is adapter with same Mac address is already registered.

x is same mac address (VF and NetVSC) merged as one interface, then later
```
outer  |--->x             |
inner  |      ----->x     |
```

```
outer  |      -----o-->     |
inner  |                    |
```
o is skipped because it was marked as Initialized earlier

Reply:
I'm still not sure why we need to keep this at all? It's not a big deal if you search the whole array N^2, since it's so small.

Reply:
Can be removed by using GAA for outer loop